packages/passport/sdk/src/Passport.ts

```diff
@@ -299,4 +303,58 @@ export class Passport {
     const getUserInfoResult = await this.multiRollupApiClients.passportProfileApi.getUserInfo({ headers });
     return getUserInfoResult.data.linked_addresses;
   }
+
+  public async linkExternalWallet(params: LinkWalletParams): Promise<LinkedWallet> {
+    track('passport', 'linkWallet', { type: params.type });
+
+    const user = await this.authManager.getUser();
+    if (!user) {
+      throw new PassportError('User is not logged in', PassportErrorType.NOT_LOGGED_IN_ERROR);
+    }
+
+    const isRegisteredWithIMX = isUserImx(user);
+    const isRegisteredWithZkEvm = isUserZkEvm(user);
+    if (!isRegisteredWithIMX && !isRegisteredWithZkEvm) {
+      throw new PassportError('User has not been registered', PassportErrorType.USER_NOT_REGISTERED_ERROR);
+    }
+
+    const headers = { Authorization: `Bearer ${user.accessToken}` };
+    const linkWalletV2Request = {
+      type: params.type,
+      wallet_address: params.walletAddress,
+      signature: params.signature,
+      nonce: params.nonce,
+    };
+
+    try {
+      const linkWalletV2Result = await this.multiRollupApiClients
+        .passportProfileApi.linkWalletV2({ linkWalletV2Request }, { headers });
+      return { ...linkWalletV2Result.data };
+    } catch (error) {
+      trackError('passport', 'linkWallet', error as Error);
+
+      if (isAxiosError(error) && error.response?.data && isAPIError(error.response.data)) {
+        const { code, message } = error.response.data;
+
+        switch (code) {
+          case 'ALREADY_LINKED':
+            throw new PassportError(message, PassportErrorType.LINK_WALLET_ALREADY_LINKED_ERROR);
+          case 'MAX_WALLETS_LINKED':
+            throw new PassportError(message, PassportErrorType.LINK_WALLET_MAX_WALLETS_LINKED_ERROR);
+          case 'DUPLICATE_NONCE':
+            throw new PassportError(message, PassportErrorType.LINK_WALLET_DUPLICATE_NONCE_ERROR);
+          case 'VALIDATION_ERROR':
+            throw new PassportError(message, PassportErrorType.LINK_WALLET_VALIDATION_ERROR);
+          default:
+            throw new PassportError(message, PassportErrorType.LINK_WALLET_GENERIC_ERROR);
+        }
+      } else {
+        // Handle unexpected error with a generic error message
+        throw new PassportError(
+          `Link wallet request failed with status code ${(error as any).response.status}`,
+          PassportErrorType.LINK_WALLET_GENERIC_ERROR,
+        );
```

Comment:
We should avoid using `any` unless we have no alternative. It also looks like this will fail if an error is thrown that doesn't have `error.response.status`. You can reproduce this by initiating a wallet link in the sample app, right click the request in dev tools and click `Block request URL`, and then attempt to link a wallet again - you'll get a `Cannot read properties of undefined (reading 'status') type: undefined` error. 

I think we can fix both of these issues by structuring the logic as follows:
```
      if (isAxiosError(error) && error.response) {
        if (error.response.data && isAPIError(error.response.data)) {
          const { code, message } = error.response.data;

          switch (code) {
            case 'ALREADY_LINKED':
              throw new PassportError(message, PassportErrorType.LINK_WALLET_ALREADY_LINKED_ERROR);
            case 'MAX_WALLETS_LINKED':
              throw new PassportError(message, PassportErrorType.LINK_WALLET_MAX_WALLETS_LINKED_ERROR);
            case 'DUPLICATE_NONCE':
              throw new PassportError(message, PassportErrorType.LINK_WALLET_DUPLICATE_NONCE_ERROR);
            case 'VALIDATION_ERROR':
              throw new PassportError(message, PassportErrorType.LINK_WALLET_VALIDATION_ERROR);
            default:
              throw new PassportError(message, PassportErrorType.LINK_WALLET_GENERIC_ERROR);
          }
        } else if (error.response.status) {
          throw new PassportError(
            `Link wallet request failed with status code ${error.response.status}}`,
            PassportErrorType.LINK_WALLET_GENERIC_ERROR,
          );
        }
      }

      let message: string = 'Link wallet request failed';
      if (error instanceof Error) {
        message += `: ${error.message}`;
      }

      throw new PassportError(
        `Link wallet request failed: ${message}`,
        PassportErrorType.LINK_WALLET_GENERIC_ERROR,
      );
```